lib/active_support/cache/memcached_store.rb

```diff
@@ -55,40 +58,34 @@ def read_multi(*names)
         keys_to_names = Hash[names.map { |name| [normalize_key(name, options), name] }]
         values = {}
 
-        instrument(:read_multi, names, options) do
-          if raw_values = @data.get(keys_to_names.keys, false)
-            raw_values.each do |key, value|
-              entry = deserialize_entry(value)
-              values[keys_to_names[key]] = entry.value unless entry.expired?
+        handle_exceptions(return_value_on_error: {}) do
+          instrument(:read_multi, names, options) do
+            if raw_values = @data.get(keys_to_names.keys, false)
+              raw_values.each do |key, value|
+                entry = deserialize_entry(value)
+                values[keys_to_names[key]] = entry.value unless entry.expired?
+              end
             end
           end
+          values
         end
-        values
-      rescue Memcached::NotFound
```

Comment:
`ConnectionDataExists` is for `add` (i.e. only add if not already there) & `cas` , I could parameterize the `handle_exceptions` for additional exceptions to be treated as "miss" though (I need to deal with `ConnectionDataExists` for cas operations. Now these two cases will (or should, i.e. that's the intent) behave exactly as they did when we swallow exceptions... I thought of adding a complete "treat these exceptions as misses" kinda param to the constructor (vs. the `:swallow_exceptions`), but it felt over the top... Willing to change my mind though.